pytket/pytket/rmbqc/knitting.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2019-2022 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from pytket.mapping import get_token_swapping_network  # type: ignore
+from pytket.circuit import Circuit, Bit, Qubit, OpType  # type: ignore
+from pytket.predicates import CompilationUnit  # type: ignore
+from pytket.passes import RoutingPass  # type: ignore
+from pytket.architecture import Architecture, FullyConnected  # type: ignore
+from typing import Dict, Tuple, List, Union
+
+
+class Knitter(Enum):
```

Comment:
please can you add more thorough comments to all these methods? Ill review properly after that